plaidml/bridge/openvino/ops/detection_output.cpp

```diff
@@ -0,0 +1,464 @@
+    }
+
+    // result = {selected_indices, selected_scores, valid_outputs}
+    edsl::Tensor iou_threshold = cast(edsl::Tensor{nms_threshold}, DType::FLOAT32);
+    edsl::Tensor score_threshold = cast(edsl::Tensor{confidence_threshold}, DType::FLOAT32);
+    std::vector<edsl::Tensor> result =
+        nms(decoded_bboxes, nms_conf, top_k, iou_threshold, score_threshold, cast(edsl::Tensor{0}, DType::FLOAT32))
+            .center_point_box(false)
+            .sort_result_descending(false)
+            .box_input_type(DType::FLOAT32)
+            .thres_type(DType::FLOAT32)
+            .build();
+    edsl::Tensor selected_indices = result[0];
+    auto selected_indices_shape = selected_indices.compute_shape().sizes();
+    edsl::Tensor selected_scores = result[1];
+    edsl::Tensor valid_outputs = result[2];
+
+    // TODO
+    edsl::Tensor batch_slice = op::slice(selected_indices).add_dim(0, selected_indices_shape[0]).add_dim(0, 1);
+    edsl::Tensor indices_slice = op::slice(selected_indices).add_dim(0, selected_indices_shape[0]).add_dim(2, 3);
+    std::vector<edsl::Tensor> gather_idxs_vec = {batch_slice, indices_slice};
+    edsl::Tensor gather_idxs = cast(op::concatenate(gather_idxs_vec, 1), DType::INT32);
+    edsl::Tensor out_boxes = edsl::gather(decoded_bboxes, gather_idxs).mode(GatherMode::ND);
+
+    if (clip_after_nms) {
+      out_boxes = op::maximum(cast(edsl::Tensor{0.0f}, DType::FLOAT32),
+                              op::minimum(cast(edsl::Tensor{1.0f}, DType::FLOAT32), out_boxes));
+    }
+
+    edsl::Tensor class_slice = op::slice(selected_indices).add_dim(0, selected_indices_shape[0]).add_dim(1, 2);
+    auto one = cast(edsl::Tensor{1}, class_slice.dtype());
+    auto slice_shape = class_slice.compute_shape().sizes();
+    std::vector<int> target_shape(begin(slice_shape), end(slice_shape));
+    std::vector<int> target_axes = {};
+    auto one_slice = op::broadcast(one, target_shape, target_axes);
+    // Update class_slice since the index of 'background_class' is removed.
+    class_slice = edsl::select(class_slice < background_label_id, class_slice, class_slice + one_slice);
+
+    edsl::Tensor scores_slice = op::slice(selected_scores).add_dim(0, selected_indices_shape[0]).add_dim(2, 3);
+
+    edsl::Tensor O = op::concatenate({batch_slice, class_slice, scores_slice, out_boxes}, 1);
+
+    edsl::Tensor topk_results;
+    if (keep_top_k[0] > -1 && selected_indices_shape[0] > keep_top_k[0]) {
+      auto sorted_idxs = op::squeeze(edsl::argsort(scores_slice, 0, edsl::SortDirection::DESC), {-1});
+      edsl::Tensor idxs_topk = edsl::gather(sorted_idxs, edsl::index({edsl::TensorDim(keep_top_k[0])}, 0));
+      auto idxs_topk_sorted = op::sort(idxs_topk, 0, edsl::SortDirection::ASC);
+      topk_results = edsl::gather(O, idxs_topk_sorted).axis(0);
+    } else {
+      // Pad -1 at the end the valid data.
+      auto neg_one = cast(edsl::Tensor{-1}, batch_slice.dtype());
+      std::vector<int> pad_shape = {1, 7};
```

Comment:
Where does `7` come from? We should avoid magic numbers as much as possible. If it is calculated by other variables please replace by them. Or you can define a new variable for 7. 
The `op::broadcast` API has changed, please use `vector<int64>` instead of `vector<int>`

Reply:
The output of `DetectionOutput` is a 7-element tuple in the form of `[batch_id, class_id, confidence, x_1, y_1, x_2, y_2]`. I will define a new descriptive variable for it.

As for the `op::broadcast` part of all comments, I will change it.